xdsl/dialects/onnx.py

```diff
@@ -26,6 +26,82 @@
 from xdsl.utils.exceptions import VerifyException
 
 
+def unidirectional_broadcast_shape(lhs: list[int], rhs: list[int]) -> list[int]:
```

Comment:
```suggestion
def unidirectional_broadcast_shape(lhs: list[int], rhs: list[int]) -> list[int] | None:
```
It feels like mixing this "business" function with the verification logic is a little confusing, the operation itself can raise the exception if the result is None